### DISCUSSION TITLE
Error Handling seems to be failing

### DISCUSSION THREAD

**Discussion by closetcodebrews** [gulpjs/gulp#2614] [Help]
Can someone help me with this:  https://stackoverflow.com/questions/68626504/error-handler-not-being-reached-for-gulp-task
I've tried to rewrite the logic to use the .catch() as well.  No complaints about syntax errors etc.  But the error handler just never kicks in. 

Please and thank you.

**Comment by phated** [comment]
This code is really bad. Why are you using gulp without tasks? Why are you wrapping gulp streams in a promise? Don't try to combine multiple types of async programming, just use one and the patterns expected by that type.

**Reply by closetcodebrews** [reply]
Thanks for the response. I'm not surprised that it's bad code.  I inherited the code and was asked to just refactor by adding add a call to sppkgDeploy when the spsync call is successful.  i modified / expanded the on("finish"). But I'm also green with JS and Gulp so that doesn't help.

In answer to your questions: 

- There are gulp tasks - just not included in the post - sorry my bad.  I only included what I thought was relevant code. 
- the author of the lib seems to suggest using pipes - https://github.com/estruyf/gulp-spsync-creds but yeah, no promises wrapped around them. 
- I'm too new to JS and gulp to know where we've mixed different types of async coding.  But that gives me somewhere to start digging / an idea of how to approach the problem.  Thanks.  Any other tips / criticism / welcome.  👯

**Reply by closetcodebrews** [reply]
Having said that, I just found this article post and it seems people are wrapping pipes with promises:  https://stackoverflow.com/questions/34188287/how-to-use-async-await-with-gulp-4

**Reply by phated** [reply]
Yep, lots of people are bad 🙃
